processing/src/main/java/org/apache/druid/query/QueryMetrics.java

```diff
@@ -396,6 +396,30 @@
    */
   QueryMetrics<QueryType> reportParallelMergeTotalCpuTime(long timeNs);
 
+  /**
+   * Reports broker total "wall" time in nanoseconds from parallel merge start sequence creation to total
+   * consumption.
+   */
+  QueryMetrics<QueryType> reportParallelMergeTotalTime(long timeNs);
+
+  /**
+   * Reports broker "wall" time in nanoseconds for the fastest parallel merge sequence partition to be 'initialized',
+   * where 'initialized' is time to the first result batch is populated from data servers and merging can begin.
+   *
+   * Similar to query 'time to first byte' metrics, except is a composite of the whole group of data servers which are
+   * present in the merge partition, which all must supply an initial result batch before merging can actually begin.
+   */
+  QueryMetrics<QueryType> reportParallelMergeFastestPartitionTime(long timeNs);
```

Comment:
1. would the slowest `ttfb` node produce slowest partition and same for the fastest? or could it be that the result batching changes the distribution of slowest and fastest partition - hence these metrics might be different from `ttfb`
2. should this method also include `initialized` in it somewhere?

Reply:
>would the slowest ttfb node produce slowest partition and same for the fastest? or could it be that the result batching changes the distribution of slowest and fastest partition - hence these metrics might be different from ttfb

yeah, these times would include the batching so its not really the first byte exactly, but they probably aren't that far off in practice. I would expect the slowest probably will most of the time be the same-ish as the slowest ttfb, though perhaps it is interesting if for some reason it isn't as it might mean the pool is overloaded or something else interesting is going on. The fastest ttfb should be faster than the fastest partition because the fastest partition will be determined by the slowest first batch of results from the servers in that partition.

Reply:
yes, agree that these metrics should correlate to TTFBs - skewness from them would be useful to find anomalies in data distribution, historical slowness or the pool being overloaded.